sdk/core/azure-core/inc/azure/core/nullable.hpp

```diff
@@ -261,10 +261,72 @@ namespace Azure { namespace Core {
       return std::move(m_value);
     }
 
+    // observers
+
     /**
      * @brief `operator bool` on the condition of #Azure::Core::Nullable::HasValue.
      */
-    explicit operator bool() const noexcept { return HasValue(); }
+    constexpr explicit operator bool() const noexcept { return HasValue(); }
+
+    /**
+     * @brief Accesses the contained value.
+     * @return Returns a pointer to the contained value
+     * @warning The behavior is undefined if `*this` does not contain a value.
+     * @note This operator does not check whether the optional contains a value!
+             You can do so manually by using #HasValue() or simply operator #bool().
+             Alternatively, if checked access is needed, #GetValue() or #ValueOr() may be used.
+     */
+    constexpr const T* operator->() const { return std::addressof(m_value); }
+
+    /**
+     * @brief Accesses the contained value.
+     * @return Returns a pointer to the contained value
+     * @warning The behavior is undefined if `*this` does not contain a value.
+     * @note This operator does not check whether the optional contains a value!
+             You can do so manually by using #HasValue() or simply operator #bool().
+             Alternatively, if checked access is needed, #GetValue() or #ValueOr() may be used.
+     */
+    constexpr T* operator->() { return _STD addressof(m_value); }
+
+    /**
+     * @brief Accesses the contained value.
+     * @return Returns a reference to the contained value
+     * @warning The behavior is undefined if `*this` does not contain a value.
+     * @note This operator does not check whether the optional contains a value!
+             You can do so manually by using #HasValue() or simply operator #bool().
+             Alternatively, if checked access is needed, #GetValue() or #ValueOr() may be used.
+     */
+    constexpr const T& operator*() const& { return m_value; }
+
+    /**
+     * @brief Accesses the contained value.
+     * @return Returns a reference to the contained value
+     * @warning The behavior is undefined if `*this` does not contain a value.
+     * @note This operator does not check whether the optional contains a value!
```

Comment:
Are we sure we want to do this? Should we do the same thing as `GetValue()` and abort if the value doesn't exist to indicate the caller to fix their calling code?

Reply:
This follows the c++ language.  Its consistent with optional.

Reply:
Seems reasonable. Thanks.
https://en.cppreference.com/w/cpp/utility/optional/operator*